cockatrice/src/gamesmodel.cpp

```diff
@@ -13,30 +13,19 @@ enum GameListColumn {ROOM, CREATED, DESCRIPTION, CREATOR, GAME_TYPE, RESTRICTION
 const QString GamesModel::getGameCreatedString(const int secs) const {
 
     QString ret;
-    if (secs < SECS_PER_MIN)
-        ret = tr("<1m ago");
-    else if (secs < SECS_PER_MIN * 5)
-        ret = tr("<5m ago");
-    else if (secs < SECS_PER_HOUR)
-        ret = tr("%1m ago").arg(QString::number(secs / SECS_PER_MIN));
-    else if (secs < SECS_PER_MIN * 90) {
-        ret = tr("1hr %1m ago").arg(QString::number((secs / SECS_PER_MIN) - 60));
-    } else if (secs < SECS_PER_HOUR * 4) {
-        unsigned int hours = secs / SECS_PER_HOUR;
-        if (secs % SECS_PER_HOUR >= SECS_PER_MIN * 30)
+    if (secs < SECS_PER_MIN * 2) // for first 2 mins
+        ret = tr("New");
+    else if (secs < SECS_PER_MIN * 30) // after 3 mins we show total mins up to 30 mins
+        ret = tr("%1 min").arg(QString::number(secs / SECS_PER_MIN));
+    else if (secs < SECS_PER_MIN * 30) { // from 30 to 1hr we show >30
```

Comment:
The duration formatting method would be a good one to unit test since it's
easy to make static. I'll see if i can add it into the pr

On Sun, Jul 10, 2016, 10:37 AM mlowe <notifications@github.com> wrote:

> In cockatrice/src/gamesmodel.cpp
> <https://github.com/Cockatrice/Cockatrice/pull/2071#discussion_r70182844>:
>
> > -    if (secs < SECS_PER_MIN)
> > -        ret = tr("<1m ago");
> > -    else if (secs < SECS_PER_MIN * 5)
> > -        ret = tr("<5m ago");
> > -    else if (secs < SECS_PER_HOUR)
> > -        ret = tr("%1m ago").arg(QString::number(secs / SECS_PER_MIN));
> > -    else if (secs < SECS_PER_MIN * 90) {
> > -        ret = tr("1hr %1m ago").arg(QString::number((secs / SECS_PER_MIN) - 60));
> > -    } else if (secs < SECS_PER_HOUR * 4) {
> > -        unsigned int hours = secs / SECS_PER_HOUR;
> > -        if (secs % SECS_PER_HOUR >= SECS_PER_MIN * 30)
> > +    if (secs < SECS_PER_MIN * 2) // for first 2 mins
> > +        ret = tr("New");
> > +    else if (secs < SECS_PER_MIN * 30) // after 3 mins we show total mins up to 30 mins
> > +        ret = tr("%1 min").arg(QString::number(secs / SECS_PER_MIN));
> > +    else if (secs < SECS_PER_MIN * 30) { // from 30 to 1hr we show >30
>
> Good find, updating.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Cockatrice/Cockatrice/pull/2071/files/fd7bc5197bae1ccfb8601fa2cec47354c92dd8cb#r70182844>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAA5NAjcYbfJ5_igPmBQMMK_ucwsUch7ks5qUQOWgaJpZM4JI1WX>
> .
>